### ISSUE TITLE
__hash__ for Languoids

### ISSUE THREAD

**Issue by antipodite** [glottolog/pyglottolog#68]
I noticed that the __hash__ magic method for Languoids is defined like so:
`
def __hash__(self):
    return id(self)
`
This means that some of the useful things you can do with sets, like `intersection`, `union` etc won't work as the hash function is operating on the memory address of the particular Languoid instance. Is this the desired behaviour, or could we change it to something like `return hash(self.glottocode)` so set operations work with Languoids?

**Comment by xrotwang** [comment]
Hm. I *think* I did that on purpose - but I can't come up with a good reason now. I'll check what changing it to - say - the Glottocode (?) will do to the tests.

**Comment by xrotwang** [comment]
@antipodite just to make sure: You would want to have `hash(self.id)` - i.e. the Glottocode - as hash for `Languoid`?

**Comment by antipodite** [comment]
Yeah, something like that, although I thought it would be better to have it hashing some pair of immutable values so the hash value of the object isn’t identical to the hash value for the glottocode as a string, but maybe it doesn’t matter

 

From: Robert Forkel ***@***.***>
Reply to: glottolog/pyglottolog ***@***.***>
Date: Monday, 27. June 2022 at 15:00
To: glottolog/pyglottolog ***@***.***>
Cc: Isaac ***@***.***>, Mention ***@***.***>
Subject: Re: [glottolog/pyglottolog] __hash__ for Languoids (Issue #68)

 

@antipodite just to make sure: You would want to have hash(self.id) - i.e. the Glottocode - as hash for Languoid?
